src/components/App.js

```diff
@@ -55,6 +55,16 @@ const App = React.createClass({
     );
   },
 
+  getInitialState() {
+    return {
+      vertical: window.matchMedia('(min-width: 700px)').matches
+    };
+  },
+
+  componentDidMount() {
+    window.matchMedia('(min-width: 700px)').onchange = event => this.setState({vertical: event.matches});
```

Comment:
I guess all this matchMedia magic should be moved to some small util lib like `KeyShortcuts` ?